### ISSUE TITLE
Parens on the LHS of products

### ISSUE THREAD

**Issue by gabrielhdt** [01mf02/kontroli-rs#12]
Hello!

## What happened

I wrote the following term in some file
```
impi: (A: Prop) -> (B: Prop) -> Prf A -> Prf B
```
and kontroli scolded me saying `Error: Parse(Term(AbstWithoutRhs))`. Since I've been initiated to the ways of the logical frameworks, I could decipher `AbstWithoutRhs`, but no abstraction were involved. Even more confused, I asked Dedukti what it thought of that term, and it was happy with it. Reading again the error, I figured out it must be that there's a syntactic construction supported by Dedukti but not Kontroli, but which one? The readme gives examples of Dedukti files, and a "Differences with Dedukti" section, but the example files don't say what's not supported, and the difference section doesn't talk about the syntax. I then tried my luck and removed the parens on the products, et voila.

## The issues, or what I'd have expected

The potential problems I see there are, in increasing order of time needed to fix it
- I couldn't find the subset of the syntax that is supported
- The error message is partly misleading
- Kontroli doesn't support the syntax `(x: A) -> B`

I could provide some documentation, but I'm not sure I could do anything for the two more complicated options.

## Environment

- kocheck 0.4.1

**Comment by 01mf02** [comment]
Oh, Monsieur Hondet! What a pleasure to read from you! (Even if it's a bug report. ^^)
Are you still working with Dedukti? What are you up to with it?

I agree, the error message is quite misleading, and your issue order is quite justified. I'll try to find some time to look into it, hopefully today.

**Comment by 01mf02** [comment]
In the code, I saw that this is a case I thought about when writing the parser:

~~~
assert_eq!(Term::parse_str("(a : b)").unwrap_err(), AbstWithoutRhs);
~~~

Such a term is expected to fail. Why? In terms, I imposed the restriction that *anything in parentheses must be a term itself*. (It probably makes the parser easier and faster, because it can then skip some special cases, like the one you encountered.) However, `A: Prop` itself is *not* a term.
